fetcher/es-fetcher/es_fetcher.go

```diff
@@ -0,0 +1,63 @@
+package fetcher
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/serhiihuberniuk/bet-predictor/fetcher/es-fetcher/es-models"
+)
+
+type ESFetcher struct {
+	client http.Client
+	token  string
+}
+
+func NewESFetcher(ctx context.Context, apiKey string) (*ESFetcher, error) {
```

Comment:
I assume that a token has lifetime and we need to re-fetch it during calls if we received an error about it's expire